fleet/lib/cnpg-cluster/fleet.yaml

```diff
@@ -17,3 +17,42 @@ dependsOn:
   - selector:
       matchLabels:
         bundle: prometheus-operator-crds
+targetCustomizations:
+  - name: pillan
+    clusterName: pillan
+    yaml:
+      overlays:
+        - pillan
+  - name: ruka
+    clusterName: ruka
+    yaml:
+      overlays:
+        - ruka
+  - name: dev
```

Comment:
`targetCustomizations` are applied as the first match starting from the start of the array.  A cluster will only match once.  As an example, `ruka` would match the `ruka` customization but never the `dev` customization as `ruka` is in the array before `dev`.

It appears that you are trying to merge the `pillan` + `tu` and `ruka` + `dev` customizations, which won't work.  A means of handling this would be to have multiple overlay dirs for each of the `pillan` and `ruka` targets.  An example of doing this (except with helm values) is https://github.com/lsst-it/k8s-cookbook/blob/master/fleet/lib/kube-prometheus-stack/fleet.yaml#L44-L56.

Reply:
since the secrets are specific for each cluster I moved the secrets inside their respective cluster and modified the fleet.yaml to only reflect pillan and ruka.